### PR TITLE
Pruning/mount functionality on model&optimizer

### PR DESCRIPTION
**Pull request by wenhuach21**
## Type of Change

another pruning API 

detail description 
mounting all the pruning functionality on the module and optimizer, users only need to call one function to enable pruning.

## Expected Behavior & Potential Risk

No

## How has this PR been tested?

unit test/ example

## Dependency Change?

No

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]

File: examples/pytorch/nlp/huggingface_models/text-classification/pruning/eager/run_glue_no_trainer_new_api.py [hunk 1]
```diff
@@ -0,0 +1,635 @@
+# coding=utf-8
```
Comment (hshen14) [inline examples/pytorch/nlp/huggingface_models/text-classification/pruning/eager/run_glue_no_trainer_new_api.py:1]: @chensuyue  please check the license usage

Comment (chensuyue) [reply examples/pytorch/nlp/huggingface_models/text-classification/pruning/eager/run_glue_no_trainer_new_api.py:1]: If we modify the code, add one copyright `# Copyright (c) 2023 Intel Corporation`

File: examples/pytorch/nlp/huggingface_models/text-classification/pruning/eager/run_glue_no_trainer.py
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/text-classification/pruning/eager/run_glue_no_trainer.py:1]: need to rollback this change later

File: examples/pytorch/nlp/huggingface_models/text-classification/pruning/eager/run_glue_no_trainer_mixed.py [hunk 2]
```diff
@@ -45,8 +45,9 @@
 )
 from transformers.file_utils import get_full_repo_name
 from transformers.utils.versions import require_version
-from neural_compressor.training import prepare_compression
-from neural_compressor.training import WeightPruningConfig
+# from neural_compressor.training import prepare_compression
+# from neural_compressor.training import WeightPruningConfig
+from neural_compressor.compression import PruningWrapper, WeightPruningConfig
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/text-classification/pruning/eager/run_glue_no_trainer_mixed.py:50]: need rollback

**Comment by wenhuach21** [comment]
readme will be updated in another pr

**Pull request merged**